### PR TITLE
use workspaces

### PR DESCRIPTION
**Pull request by jquense**
Uses yarn workspaces! gatsby dev now requires yarn ^1.0.2

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 36e2feda9b279e39ee349e781b71210e22c5a3d8

https://app.netlify.com/sites/using-glamor/deploys/59d28095a6188f0bce89ea2e

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit c0da6a14c54078d8aa067b67aac4bd1f665bd5a4

https://app.netlify.com/sites/using-glamor/deploys/59d2809edf99535b39fb1d18

File: packages/gatsby-source-filesystem/index.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"use strict";
```
Comment (KyleAMathews) [inline packages/gatsby-source-filesystem/index.js]: For some reason the .gitignore file isn't ignoring this

File: package.json [hunk 2]
```diff
@@ -48,7 +51,7 @@
     "lint": "npm run check-versions && eslint --ext .js,.jsx --ignore-path .gitignore .",
     "lint:flow": "babel-node scripts/flow-check.js",
     "plop": "plop",
-    "bootstrap": "yarn && npm run check-versions && lerna bootstrap",
+    "bootstrap": "yarn && npm run check-versions && lerna run prepublish",
```
Comment (KyleAMathews) [inline package.json]: Do we want to run prepublish for bootstrapping? Perhaps this is a problem just with my workflow but I'm always watching the repo to do changes so don't need to compile files as part of bootstrap. Or does prepublish do more than just compile now?

Though probably it makes sense as the default. With workspaces I can just run `yarn` now to install everything everywhere right?

Comment (jquense) [reply package.json]: It's safe thing to do, and required for tests to pass. The issue is on packages that depend on each other, if `gatsby-foo` needs `gatsby-foo` jest isn't going to resolve `gatsby-foo` to the `src` directory so the compiled files need to exist. It's not strictly required to get going tho.  You can just run `yarn` and start working and everything will be installed 👍

**Comment by KyleAMathews** [comment]
Thanks!

**Pull request merged**